### ISSUE TITLE
Traverser examples in Kotlin

### ISSUE THREAD

**Issue by amatiushkin** [intuit/Traverser#17]
There few things which would be nice to have as part of this PR:
- example of how to import Traverser into Kotlin project 
- sample code snippet which runs DFS/BFS pre/post order against example data set (for example company organization chart or bring your sample)

As one of practical suggestion -- StackOverflow or some other forum might have a question about traversing something in mobile app, this could be a good tasks which can be solved with traversal as well.

**Comment by abhishekganesan** [comment]
@amatiushkin

**Comment by shelley-tong** [comment]
Is this assigned to anyone yet? if not I'm interested

**Comment by qbool** [comment]
Yes it's assigned to me. Although you can implement these using Scala see #16

**Comment by stong2351** [comment]
Cool, I just responded to that thread. I'd be willing to use Scala

**Comment by vaggarwal2** [comment]
I would like to add it in Kotlin.

**Comment by vaggarwal2** [comment]
Looks like a PR is already raised for it. I can work in Groovy.
